spec/toc_spec.rb

```diff
@@ -1,64 +1,396 @@
-#NOTE: start the middleman as in readme, then run bundle exec rspec
+require "spec_helper"
 
-require 'capybara/rspec'
-require 'capybara/poltergeist'
-require 'pry'
+describe Highlighter::Helpers do
+  let(:helper)                 { HelperTester.new }
+  let(:building_page)          { double(path: "custom-extensions/building-custom-extensions.html") }
+  let(:testing_page)           { double(path: "custom-extensions/testing-custom-extensions.html") }
+  let(:building_chapter_title) { "Building Custom Extensions" }
+  let(:building_chapter_url)   { "building-custom-extensions" }
+  let(:testing_chapter_title)  { "Testing Custom Extensions" }
+  let(:testing_chapter_url)    { "testing-custom-extensions" }
+  let(:building_chapter)       { double(title: building_chapter_title, url: building_chapter_url) }
+  let(:testing_chapter)        { double(title: testing_chapter_title, url: testing_chapter_url) }
+  let(:guide_title)            { "Custom Extensions" }
+  let(:guide_url)              { "custom-extensions" }
+  let(:guide)                  { double(title: guide_title, url: guide_url, chapters: [building_chapter, testing_chapter]) }
+  let(:test_data)              { double(guides: [guide]) }
 
-unless `which phantomjs`.empty?
-  Capybara.register_driver :poltergeist do |app|
-    Capybara::Poltergeist::Driver.new(app, { :inspector => true, })
+  before(:each) do
+    class HelperTester
+      include TOC::Helpers
+    end
+
+    allow(helper).to receive(:current_page).and_return(testing_page)
```

Comment:
`current_page` is not something that we override in the helper and it felt was not in violation of mocking something we don't own. Is the issue that `current_page` is mocked out in a `before` block instead of each individual `example`?